main.go

```diff
@@ -343,19 +344,15 @@ func mainerr() error {
 			case *fVersion:
 				fmt.Printf("%v %v\n", mp.Module.Path, mp.Module.Version)
 			case *fRun:
-				if runtime.GOOS == "windows" {
-					run := exec.Command(target, runArgs...)
-					run.Stdout = os.Stdout
-					run.Stderr = os.Stderr
-					if err := run.Run(); err != nil {
-						if _, ok := err.(*exec.ExitError); !ok {
-							return fmt.Errorf("failed to run %v: %v", strings.Join(run.Args, " "), err)
-						}
-					}
-				} else {
-					argv := append([]string{target}, runArgs...)
-					if err := syscall.Exec(argv[0], argv, os.Environ()); err != nil {
-						return fmt.Errorf("failed to exec %v: %v", strings.Join(argv, " "), err)
+				run := exec.Command(target, runArgs...)
+				run.Args[0] = path.Base(mp.ImportPath)
+				run.Stdout = os.Stdout
+				run.Stderr = os.Stderr
+				if err := run.Run(); err != nil {
+					if _, ok := err.(*exec.ExitError); ok {
+						return errRunExitErr
```

Comment:
Looks like this is discarding the value of the exit code. Might it be a good idea to preserve it?